src/serde.rs

```diff
@@ -0,0 +1,267 @@
+        state.serialize_field("virtual_memory", &self.virtual_memory())?;
+        if let Some(pid) = self.parent() {
+            state.serialize_field("parent", &pid.as_u32())?;
+        }
+        state.serialize_field("status", &self.status())?;
+        state.serialize_field("start_time", &self.start_time())?;
+        state.serialize_field("run_time", &self.run_time())?;
+        state.serialize_field("cpu_usage", &self.cpu_usage())?;
+        state.serialize_field("disk_usage", &self.disk_usage())?;
+        if let Some(uid) = self.user_id() {
+            let uid = *uid.deref();
+            state.serialize_field("user_id", &uid)?;
+        }
+        if let Some(gid) = self.group_id() {
+            let gid = *gid.deref();
+            state.serialize_field("group_id", &gid)?;
+        }
+        if let Some(pid) = self.session_id() {
+            state.serialize_field("session_id", &pid.as_u32())?;
+        }
+
+        state.end()
+    }
+}
+
+impl Serialize for dyn CpuExt {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut state = serializer.serialize_struct("Cpu", 1)?;
+
+        state.serialize_field("cpu_usage", &self.cpu_usage())?;
+        state.serialize_field("name", &self.name())?;
+        state.serialize_field("vendor_id", &self.vendor_id())?;
+        state.serialize_field("brand", &self.brand())?;
+        state.serialize_field("frequency", &self.frequency())?;
+
+        state.end()
+    }
+}
+
+impl serde::Serialize for dyn SystemExt {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        let mut state = serializer.serialize_struct("System", 27)?;
+
+        // IS_SUPPORTED: bool
+        // SUPPORTED_SIGNALS: &'static [Signal]
+        // MINIMUM_CPU_UPDATE_INTERVAL: Duration;
+        // &self.global_cpu_info() -> &Cpu;
+        // &self.cpus -> &[Cpu];
+
+        state.serialize_field("physical_core_count", &self.physical_core_count())?; // Option<usize>
+        state.serialize_field("total_memory", &self.total_memory())?; // u64
+        state.serialize_field("free_memory", &self.free_memory())?; // u64
+        state.serialize_field("available_memory", &self.available_memory())?; // u64
+        state.serialize_field("used_memory", &self.used_memory())?; // u64
+        state.serialize_field("total_swap", &self.total_swap())?; // u64
+        state.serialize_field("free_swap", &self.free_swap())?; // u64
+        state.serialize_field("used_swap", &self.used_swap())?; // u64
+
+        // &self.components() -> &[Component]
+        // &self.users() -> &[User]
+        // &self.disks() -> &[Disk]
+        // &self.networks() -> &Networks
+
+        state.serialize_field("uptime", &self.uptime())?; // u64
+        state.serialize_field("boot_time", &self.boot_time())?; // u64
+
+        // &self.load_average() -> LoadAvg
```

Comment:
I think you can add serialization for this one.

Reply:
well, the fact that we are doing it kinda sends the opposite message.

Reply:
You can cast your variable with `as dyn trait` without problem, that's how I tested it.